music_assistant/providers/podcastfeed/__init__.py

```diff
@@ -0,0 +1,283 @@
+        Setting this to False will query all instances of this provider for search and lookups.
+        """
+        return False
+
+    async def search(
+        self,
+        search_query: str,
+        media_types: list[MediaType],
+        limit: int = 5,
+    ) -> SearchResults:
+        """Perform search on musicprovider.
+
+        :param search_query: Search query.
+        :param media_types: A list of media_types to include.
+        :param limit: Number of items to return in the search (per type).
+        """
+        result = SearchResults()
+
+        if MediaType.PODCAST in media_types or media_types is None:
+            # return podcast if artist matches podcast name
+            if search_query in self.parsed["title"]:
+                result.podcasts.append(await self._parse_podcast())
+
+        # if MediaType.EPISODE in media_types or media_types is None:
+        #    if search_query in self.parsed["title"]:
+        #        for episode in self.parsed["episodes"]:
+        #            result.podcasts.append(await self._parse_episode(episode))
+
+        return result
+
+    async def get_library_podcasts(self) -> AsyncGenerator[Podcast, None]:
+        """Retrieve library/subscribed podcasts from the provider."""
+        yield await self._parse_podcast()
+
+    async def get_podcast(self, prov_podcast_id: str) -> Podcast:
+        """Get full artist details by id."""
+        return await self._parse_podcast()
+
+    # type: ignore[return]
+    async def get_episode(self, prov_episode_id: str) -> Episode:
+        """Get (full) podcast episode details by id."""
+        for episode in self.parsed["episodes"]:
+            if prov_episode_id in episode["guid"]:
+                return await self._parse_episode(episode)
+        raise MediaNotFoundError("Track not found")
+
+    async def get_podcast_episodes(
+        self,
+        prov_podcast_id: str,
+    ) -> list[Episode]:
+        """List all episodes for the podcast."""
+        episodes = []
+
+        for episode in self.parsed["episodes"]:
+            episodes.append(await self._parse_episode(episode, prov_podcast_id))
+
+        return episodes
+
+    async def library_add(self, item: MediaItemType) -> bool:
+        """Add item to provider's library. Return true on success."""
+        return True
+
+    async def library_remove(self, prov_item_id: str, media_type: MediaType) -> bool:
+        """Remove item from provider's library. Return true on success."""
+        return True
+
+    async def get_stream_details(self, item_id: str) -> StreamDetails:
+        """Get streamdetails for a track/radio."""
+        for episode in self.parsed["episodes"]:
+            if item_id in episode["guid"]:
+                return StreamDetails(
+                    provider=self.instance_id,
+                    item_id=item_id,
+                    audio_format=AudioFormat(
+                        # hard coded mp3 for now
```

Comment:
If you are unsure about the content type, you may set this to unknown. This will instruct ffmpeg in the streaming chain to figure out the codec itself.